File: flutter_contactless_sdk/ios/Classes/ContactlessPlugin.swift
import Flutter
import UIKit
import CoreNFC

public class ContactlessPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "com.contactless/sdk", binaryMessenger: registrar.messenger())
    let instance = ContactlessPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initialize":
      // Initialize security/Apple Pay integration
      result(true)
    case "isNfcAvailable":
      result(NFCTagReaderSession.readingAvailable)
    case "startPayment":
      // Trigger CoreNFC or PassKit
      // Mocking success response
      let mockResponse: [String: Any] = [
        "success": true,
        "transactionId": "TXN_\(Int(Date().timeIntervalSince1970))",
        "authCode": "654321",
        "maskedPan": "5412********4444"
      ]
      result(mockResponse)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: flutter_contactless_sdk/android/src/main/kotlin/com/contactless/ContactlessPlugin.kt
package com.contactless

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import android.content.Context
import android.nfc.NfcAdapter

/** ContactlessPlugin */
class ContactlessPlugin: FlutterPlugin, MethodCallHandler {
  private lateinit var channel : MethodChannel
  private lateinit var context: Context
  private var nfcAdapter: NfcAdapter? = null

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.contactless/sdk")
    channel.setMethodCallHandler(this)
    context = flutterPluginBinding.applicationContext
    nfcAdapter = NfcAdapter.getDefaultAdapter(context)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    when (call.method) {
      "initialize" -> {
        val apiKey = call.argument<String>("apiKey")
        val environment = call.argument<String>("environment")
        // Initialize security modules here
        result.success(true)
      }
      "isNfcAvailable" -> {
        result.success(nfcAdapter?.isEnabled ?: false)
      }
      "startPayment" -> {
        val amount = call.argument<Double>("amount")
        val currency = call.argument<String>("currency")
        
        // This would trigger the NFC session and EMV processing
        // For now, we return a mock success
        val mockResponse = mapOf(
          "success" to true,
          "transactionId" to "TXN_" + System.currentTimeMillis(),
          "authCode" to "123456",
          "maskedPan" to "4111********1111"
        )
        result.success(mockResponse)
      }
      else -> {
        result.notImplemented()
      }
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
